engine/src/main/java/graphql/nadel/engine/execution/ServiceResultNodesToOverallResult.java

```diff
@@ -594,12 +590,46 @@ public TraversalControl enter(TraverserContext<ExecutionResultNode> context) {
         });
     }
 
-    private List<String> getFieldIdsWithoutTransformationId(ExecutionResultNode node, TransformationMetadata transformationMetadata) {
-        return node.getFieldIds().stream().filter(fieldId -> FieldMetadataUtil.getTransformationIds(fieldId, transformationMetadata.getMetadataByFieldId()).size() == 0).collect(Collectors.toList());
+    private ExecutionResultNode nodesWithoutTransformationIds(ExecutionResultNode executionResultNode, Set<String> transformationIds, TransformationMetadata transformationMetadata) {
+
+        return resultNodesTransformer.transform(executionResultNode, new TraverserVisitorStub<ExecutionResultNode>() {
+
+            @Override
+            public TraversalControl enter(TraverserContext<ExecutionResultNode> context) {
+                ExecutionResultNode node = context.thisNode();
+                List<String> fieldIdsWithId;
+
+                fieldIdsWithId = getFieldIdsWithoutTransformationIds(node, transformationIds, transformationMetadata);
+
+
+                if (fieldIdsWithId.isEmpty()) {
+                    return TreeTransformerUtil.deleteNode(context);
+                }
+                ExecutionResultNode changedNode = changeFieldIsInResultNode(node, fieldIdsWithId);
+                return TreeTransformerUtil.changeNode(context, changedNode);
+            }
+        });
     }
 
-    private List<String> getFieldIdsWithTransformationIds(ExecutionResultNode
-                                                                  node, Set<String> transformationIds, TransformationMetadata transformationMetadata) {
+    private List<String> getFieldIdsWithoutTransformationIds(ExecutionResultNode node, Set<String> notTransformationIds, TransformationMetadata transformationMetadata) {
+        return FpKit.filter(node.getFieldIds(), fieldId -> {
+            Map<String, List<FieldMetadata>> metadataByFieldId = transformationMetadata.getMetadataByFieldId();
+            List<String> transformationIds = FieldMetadataUtil.getTransformationIds(fieldId, metadataByFieldId);
```

Comment:
Q: In which scenario would `getTransformationIds` return ids that are not present in `allRelevantTransformationIds` which is passed down as an argument to this method? Why just checking that the collection returned from `getTransformationIds` is empty (which was what the method was doing before) not enough to assert that the field doesn't have transformation ids associated with it?